crates/cheatcodes/src/inspector.rs

```diff
@@ -1530,11 +1530,14 @@ impl Cheatcodes {
             return None;
         }
 
-        if let TransactTo::Call(test_contract) = ecx.env.tx.transact_to {
-            if call.bytecode_address == test_contract {
-                info!("running call in EVM, instead of zkEVM (Test Contract) {:#?}", ecx.env.tx);
-                return None
-            }
+        if let Some(true) =
+            ecx.db.get_test_contract_address().map(|addr| call.bytecode_address == addr)
+        {
+            info!(
+                "running call in EVM, instead of zkEVM (Test Contract) {:#?}",
+                call.bytecode_address
+            );
+            return None
```

Comment:
I'm sure there was an issue why I couldn't simply check for this value being populated, maybe in setup or something. I'd like to run some tests here to ensure we're not introducing regression.

I can take care of that.